### PR TITLE
Docs - revamp README

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
This PR aims to improve our docs landing page, the [README](https://www.metabase.com/docs/latest/):

### Revamps the README

- Adds a lot more info.
- Reduces word count and groups items into sections to make the list easier to scan.
- Groups content by use case rather than persona (though it retains the guides). The idea here is to surface functionality, rather than hide it under packages like "Users guide".

### Updates the Enterprise page

- Incorporates Pro plans.
- Adds some sections and links.

### Updates Operations guide

- Adds sections.

## What this PR doesn't do

It doesn't change the docs file/directory structure. The README is a list of links to curate existing content regardless of where the content lives. For example, documents covering permissions were scattered in different places and different guides; this PR groups them in one section. We can consider changing directory structure later on, but that will involve some heavy lifting managing redirects in the website repo that serves these pages (which could include rewriting the code that decorates these pages with frontmatter).

## Concerns

- Some of our docs have a sequential Next link at the bottom to take them to the next article in the sequence. The groupings don't respect those sequences, which could cause some confusion. We've already broken away from this sequencing lock-in as we've added more docs, but we should consider rooting out these vestigial "Next" sections.
- The README is long. I included the brutalist version with links to all pages because it's still manageable and people might want to skim and/or ctrl-f through the page, but we could also consider a shorter version with landing pages for subsections. E.g., instead of a Permissions section that lists all of the links in the README, Permissions is just one item on the README list that will take you to a landing page on Permissions that lists the subpages.

## Next steps

If/once this PR is merged, we should consider:

- Going through and marking existing pages in the Enterprise guide with the plans blockquote, so people don't have to know they're in the Enterprise guide to understand what they're looking at describes a paid feature.
- (Maybe) starring or somehow marking items in the main README list to indicate they're paid features. 
- Consolidating some pages with related content. For example, the page that talks about [customizing filter values for dashboard subscriptions](https://www.metabase.com/docs/latest/enterprise-guide/dashboard-subscriptions.html) should probably just be a section on the dashboard subscription page rather than a standalone note. Their current split is an artifact of our current (awkward) demarcation of paid/free feature this PR sets up to correct.
- Folding the FAQ content into relevant contexts and delete the FAQ.
- Condense the line height on lists so they're easier to skim.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,103 +1,168 @@
...
 
-A place to get help on installation, setting up as well as sharing tips and tricks.
+- [API reference][api-documentation]
+- [API tutorial][api-tutorial]
 
-### [Data Bytes][data-bytes]
+## Troubleshooting and getting help
 
-Real stories about teams working and learning with data. You can also share your own stories.
+- [Troubleshooting Guide][troubleshooting].
```
Comment (frannietrempe) [inline docs/README.md:117]:
```suggestion
- [Troubleshooting guide][troubleshooting] 
```

File: docs/README.md [hunk 2]
```diff
@@ -1,103 +1,168 @@
...
-A place to get help on installation, setting up as well as sharing tips and tricks.
+- [API reference][api-documentation]
+- [API tutorial][api-tutorial]
 
-### [Data Bytes][data-bytes]
+## Troubleshooting and getting help
 
-Real stories about teams working and learning with data. You can also share your own stories.
+- [Troubleshooting Guide][troubleshooting]. 
+- [Metabase forum][forum].
```
Comment (frannietrempe) [inline docs/README.md:118]:
```suggestion
- [Metabase forum][forum]
```

File: docs/README.md [hunk 3]
```diff
@@ -1,103 +1,168 @@
...
+- [Admin guide](administration-guide/start.html)
+- [Operations guide](operations-guide/start.html)
+- [Troubleshooting guide][troubleshooting] 
+- [Developers guide][developers]
 
 ## Reference
 
-### [Anonymous Information Collection Reference][info-collection]
+- [Anonymous Information Collection Reference][info-collection]
+- [FAQs][faq]
```
Comment (frannietrempe) [inline docs/README.md:145]:
```suggestion
- [FAQs][faq]
- [Glossary][glossary]
```

File: docs/README.md [hunk 4]
```diff
@@ -1,103 +1,168 @@
...
 [data-bytes]: /community
-[developers]: developers-guide/start.md
-[drivers]: developers-guide-drivers.md
-[faq]: faq/start.md
+[developers]: developers-guide/start.html
+[drivers]: developers-guide-drivers.html
+[faq]: faq/start.html
 [forum]: https://discourse.metabase.com/
-[info-collection]: information-collection.md
+[getting-started]: /learn/getting-started/getting-started.html
```
Comment (frannietrempe) [inline docs/README.md:160]:
```suggestion
[getting-started]: /learn/getting-started/getting-started.html
[glossary]: /glossary.html
```

File: docs/README.md [hunk 5]
```diff
@@ -1,103 +1,168 @@
...
+- [API reference][api-documentation]
+- [API tutorial][api-tutorial]
 
-### [Data Bytes][data-bytes]
+## Troubleshooting and getting help
 
-Real stories about teams working and learning with data. You can also share your own stories.
+- [Troubleshooting Guide][troubleshooting]. 
+- [Metabase forum][forum].
+- [Configuring Logging](./operations-guide/log-configuration.html)
```
Comment (frannietrempe) [inline docs/README.md:119]:
```suggestion
- [Configuring logging](./operations-guide/log-configuration.html)
```

File: docs/operations-guide/start.md [hunk 6]
```diff
@@ -1,26 +1,35 @@
...
+-  [Changing password complexity](changing-password-complexity.html)
+-  [Changing session expiration](changing-session-expiration.html)
+-  [Handling Timezones](handling-timezones.html)
+-  [Configuring Logging](log-configuration.html)
+-  [How to setup monitoring via JMX](jmx-monitoring.html)
+-  [A word on Java versions](java-versions.html)
+
+## Scaling Metabase
+
+- [Metabase at Scale](https://www.metabase.com/blog/scaling-metabase/index.html)
```
Comment (frannietrempe) [inline docs/operations-guide/start.md:35]:
```suggestion
- [Metabase at scale](https://www.metabase.com/blog/scaling-metabase/index.html)
```

File: docs/operations-guide/start.md [hunk 7]
```diff
@@ -1,26 +1,35 @@
...
+-  [Configuring settings using environment variables](environment-variables.html)
+-  [Connecting Metabase to databases in your organization](../administration-guide/01-managing-databases.html)
+-  [Encrypting your database connection details at rest](encrypting-database-details-at-rest.html)
+
+## Configuring Metabase
+
+-  [Customizing the Metabase Jetty Webserver](customizing-jetty-webserver.html)
+-  [Changing password complexity](changing-password-complexity.html)
+-  [Changing session expiration](changing-session-expiration.html)
+-  [Handling Timezones](handling-timezones.html)
```
Comment (frannietrempe) [inline docs/operations-guide/start.md:28]:
```suggestion
-  [Handling timezones](handling-timezones.html)
```

File: docs/operations-guide/start.md [hunk 8]
```diff
@@ -1,26 +1,35 @@
...
+-  [Configuring Logging](log-configuration.html)
```
Comment (frannietrempe) [inline docs/operations-guide/start.md:29]:
```suggestion
-  [Configuring logging](log-configuration.html)
```

**Review by frannietrempe — APPROVED** [review]
Looks great. 🥳

**Comment by jeff-bruemmer** [comment]
@metabase-bot backport release-x.41.x

**Pull request merged**